server/app/src/mocap_server.py

```diff
@@ -642,56 +609,25 @@ def updateStatusFile(self):
                 'protected/server_status/'+self.serverId, statusStr)
             print('Uploaded updated status file: \n' + statusStr)
 
-    def cleanUpOtherDeadServersForever(self):
-        """
-        This will spin, pinging anyone else whose server status files are still online at intervals, 
-        and cleaning up servers that haven't responded for more than a configured timeout.
-        """
-        pingIntervalSeconds = 30
-        deathIntervalSeconds = 60
+    def onPubSubStatusReceived(self, topic: str, payload: bytes):
+        print(f'Received PubSub status update on server {self.serverId}')
+        self.index.pubSub.alive = True
+
+    def checkPubSubStatusForever(self):
 
         while True:
-            statusFiles: Dict[str, FileMetadata] = self.index.getImmediateChildren(
-                "protected/server_status/")
+            # First, assume that PubSub is down.
+            self.index.pubSub.alive = False
 
-            # Send out pings
-            for k in statusFiles:
-                if k == self.serverId:
-                    pass
-                else:
-                    # Send a ping, which will get an asynchronous response which will eventually update self.lastSeenPong[k]
-                    try:
-                        self.index.pubSub.sendMessage('/PING/'+k, {})
-                    except e:
-                        print('Failed to send ping to ' +
-                              k+': '+str(e), flush=True)
+            # Send a status update message and wait a few seconds.
+            self.index.pubSub.publish('/PUBSUB_STATUS/'+self.serverId, {})
+            time.sleep(5)
 
-            # Check for death
-            for k in statusFiles:
-                if k == self.serverId:
-                    pass
-                else:
-                    if k not in self.lastSeenPong:
-                        self.lastSeenPong[k] = time.time()
-                    timeSincePong = time.time() - self.lastSeenPong[k]
-                    if timeSincePong > deathIntervalSeconds:
-                        print('Detected dead server: '+str(k), flush=True)
-                        self.index.delete('protected/server_status/'+k)
-                        print('Cleaned up status file for: '+str(k), flush=True)
-                    else:
-                        print('Server '+str(k)+' seen within '+str(timeSincePong) +
-                              's < death interval '+str(deathIntervalSeconds)+'s, so still alive', flush=True)
-
-            time.sleep(pingIntervalSeconds)
+            # If we didn't get a response, then PubSub is down.
+            if not self.index.pubSub.alive:
+                print('PubSub is down!')
```

Comment:
This logic makes sense. I assume we're going to have a local flag in mocap_server for PubSub being down, and pause processing, and then resume if PubSub comes back? It's probably not worth exiting the whole process when this happens, because it happens ofter, and it's generally only out for a minute or less.

Reply:
In `processQueueForever` below, I added the "alive" flag as a requirement for more jobs to be queued. I think this should work since, even if PubSub goes down during the middle of a job queue, the loop will catch the PubSub "alive" status on the next go around and not try to re-queue the same job.

Reply:
Is this going to break some percentage of the time because every 5 seconds we set `alive` to false, and then it stays false for a while until we get the response from PubSub (say perhaps 500ms, so 10% of the 5s cycle time)? Should this logic use another flag that doesn't get set to false unless PubSub actually goes down?